### PR TITLE
Research 939 jetson

### PR DESCRIPTION
**Pull request by sebatyler**
- Define a public docker for jetson

### CODE REVIEW

**Comment by jahwanoh** [comment]
@bepro-company/camera-developers , Could you proceed with the review?

- adding opencv installation script.
- adding tensorflow prebuilt package script (when scouting feed with TensorRT is ready, I will remove this.)
- Dockerfile for arm64 platform(Jetson)

File: stitching_base/Dockerfile.arm64 [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+RUN bash nvcodec.sh && rm -rfv nvcodec.sh nv-codec-headers
+
+ADD tensorflow_prebuilt.sh .
+RUN bash tensorflow_prebuilt.sh && rm -rfv tensorflow_prebuilt.sh 
+
+ADD opencv.sh.arm64 ./opencv.sh
+RUN bash opencv.sh && rm -rfv opencv.sh
+
+ADD live555.sh .
+RUN bash live555.sh && rm -rfv live555.sh live
```
Comment (sebatyler) [inline stitching_base/Dockerfile.arm64]: Removing live directory should be removed like this PR.
https://github.com/bepro-company/docker/pull/21/files

File: stitching_base/Dockerfile.arm64 [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+       sudo g++-7 gdb clang-format-10 libgstreamer1.0-dev libgstreamer-plugins-base1.0-dev libgtk2.0-dev libtbb-dev \
+       qt5-default libatlas-base-dev libfaac-dev libtheora-dev libvorbis-dev libxvidcore-dev libopencore-amrnb-dev \
+       libopencore-amrwb-dev x264 v4l-utils libva-dev libx11-dev xorg-dev libglew1.5 libglew1.5-dev \
+       libglu1-mesa libglu1-mesa-dev libgl1-mesa-glx libgl1-mesa-dev libglfw3-dev libglfw3 freeglut3-dev \
+       libzbar-dev libzbar0 libmediainfo-dev mediainfo libqrencode-dev \
+      python3-gst-1.0 && \
+      apt-get install --reinstall ca-certificates && \
+      rm -rf /var/lib/apt/lists/* && \
+      # use latest cmake
+      pip3 install -U pip && \
```
Comment (sebatyler) [inline stitching_base/Dockerfile.arm64]: What about fixing pip version like this?
```bash
pip3 install pip==21.3.1
```

File: stitching_base/tensorflow_prebuilt.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+#tensorflow cc
+cd ~
+wget https://storage.googleapis.com/bepro-dev/scouting_feed/tensorflow/tensorflow_cpp_jetson_2.3.1.zip
+mkdir tensorflow_cc
+cd tensorflow_cc
+unzip ../tensorflow_cpp_jetson_2.3.1.zip
+cp -r include/* /usr/local/include
+
+rm libtensorflow_cc.so
+rm libtensorflow_cc.so.2
```
Comment (sebatyler) [inline stitching_base/tensorflow_prebuilt.sh]: Removing files can be merged.
```
rm -v libtensorflow_cc.so libtensorflow_cc.so.2 libtensorflow_framework.so.2
```

File: stitching_base/tensorflow_prebuilt.sh [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+ln -s libtensorflow_cc.so.2.3.1 libtensorflow_cc.so
+ln -s libtensorflow_cc.so.2.3.1 libtensorflow_cc.so.2
+rm libtensorflow_framework.so.2
+ln -s libtensorflow_framework.so.2.3.1 libtensorflow_framework.so.2
+
+cp libtensorflow_* /usr/local/lib
+cd ~
+rm -r tensorflow_cc
+rm tensorflow_cpp_jetson_2.3.1.zip
```
Comment (sebatyler) [inline stitching_base/tensorflow_prebuilt.sh]: These 2 lines also can be merged.
```
rm -rfv tensorflow_cc tensorflow_cpp_jetson_2.3.1.zip
```

File: stitching_base/tensorflow_prebuilt.sh [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+cp -r include/* /usr/local/include
+
+rm libtensorflow_cc.so
+rm libtensorflow_cc.so.2
+ln -s libtensorflow_cc.so.2.3.1 libtensorflow_cc.so
+ln -s libtensorflow_cc.so.2.3.1 libtensorflow_cc.so.2
+rm libtensorflow_framework.so.2
+ln -s libtensorflow_framework.so.2.3.1 libtensorflow_framework.so.2
+
+cp libtensorflow_* /usr/local/lib
```
Comment (sebatyler) [inline stitching_base/tensorflow_prebuilt.sh]: What about using -v option to display copying files?

File: stitching_base/tensorflow_prebuilt.sh [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+cd ~
+rm protobuf-cpp-3.9.2.zip
+
+#tensorflow cc
+cd ~
+wget https://storage.googleapis.com/bepro-dev/scouting_feed/tensorflow/tensorflow_cpp_jetson_2.3.1.zip
+mkdir tensorflow_cc
+cd tensorflow_cc
+unzip ../tensorflow_cpp_jetson_2.3.1.zip
+cp -r include/* /usr/local/include
```
Comment (sebatyler) [inline stitching_base/tensorflow_prebuilt.sh]: -v option is also helpful here.

File: stitching_base/tensorflow_prebuilt.sh (hunk 3 above)
Comment (jahwanoh) [reply stitching_base/tensorflow_prebuilt.sh]: Thank you I will put -v option and write in one line as much as possible!

**Comment by jahwanoh** [comment]
Thank you for the review, I applied the comments.

File: stitching_base/tensorflow_prebuilt.sh [hunk 7]
```diff
@@ -0,0 +1,28 @@
+#protobuff
+cd ~
+wget https://github.com/protocolbuffers/protobuf/releases/download/v3.9.2/protobuf-cpp-3.9.2.zip
+unzip protobuf-cpp-3.9.2.zip
+cd protobuf-3.9.2/cmake
+mkdir build && cd build
+cmake ..
+make -j8
```
Comment (sebatyler) [inline stitching_base/tensorflow_prebuilt.sh]: `make -j$(nproc)` would be better. Because nproc tells how many cores there are.

Comment (jahwanoh) [reply stitching_base/tensorflow_prebuilt.sh]: 수정했습니다!

**Comment by sebatyler** [comment]
All confirmed. But I couldn't approve because I made this PR. Please approve by yourself 😄

**Comment by sebatyler** [comment]
@jahwanoh Please check my last comment!

**Review by nalssing — APPROVED** [review]
Then I will approve too 💯

**Pull request merged**